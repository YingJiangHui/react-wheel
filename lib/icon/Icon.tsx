import React, {FunctionComponent} from 'react';
import './importIcons';
import './icon.scss';
import classes from '../helper/classes'
interface Props extends React.HTMLAttributes<SVGElement>{
  name: string
}

const Icon: FunctionComponent<Props> = ({className,name,...restProps}) => {
  return (
    <svg className={classes('xxx-icon',className)} {...restProps}>
      <use xlinkHref={'#' + name} />
    </svg>
  );
};
export default Icon;