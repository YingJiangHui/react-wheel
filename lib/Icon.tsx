import React, {FunctionComponent} from 'react';
import './importIcons';
import './icon.scss';

interface Props {
  name: string
  onClick?:React.MouseEventHandler<SVGElement>
}

const Icon: FunctionComponent<Props> = (props) => {
  return (
    <svg className={'xxx-icon'} onClick={props.onClick}>
      <use xlinkHref={'#' + props.name}></use>
    </svg>
  );
};
export default Icon;