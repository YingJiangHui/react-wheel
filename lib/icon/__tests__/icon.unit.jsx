import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../Icon'

describe('测试icon',()=>{
  it('icon代码快照是否正确',()=>{
    const json = renderer.create(<Icon name={'pay'}/>).toJSON()
    expect(json).toMatchSnapshot()
  })
})