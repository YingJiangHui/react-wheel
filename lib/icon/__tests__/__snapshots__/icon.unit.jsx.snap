// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试icon icon代码快照是否正确 1`] = `
<svg
  className="makabaka-icon"
>
  <use
    xlinkHref="#pay"
  />
</svg>
`;
